src/push-notifications.js

```diff
@@ -49,16 +49,23 @@ export class Client {
   }
 
   async start() {
-    const { vapidPublicKey: publicKey } = await this._getPublicKey();
-
-    // register with pushManager, get endpoint etc
-    const token = await this._getPushToken(publicKey);
-
-    // get device id from errol
-    const response = await this._registerDevice(token);
-    this.deviceId = response;
-
-    this._save(this.instanceId, token, this.deviceId);
+    const data = this._read(this.instanceId);
+    if (typeof data !== 'undefined' && data.length > 0) {
+      this.token = data[0];
+      this.deviceId = data[1];
+      return;
+    } else {
+      const { vapidPublicKey: publicKey } = await this._getPublicKey();
```

Comment:
If we don't have values stored in the database, do the request to the server, ...